tests/js/spec/utils/useApi.spec.tsx

```diff
@@ -0,0 +1,58 @@
+import {mountWithTheme} from 'sentry-test/reactTestingLibrary';
+
+import {Client} from 'app/api';
+import useApi from 'app/utils/useApi';
+
+describe('useApi', function () {
+  let apiInstance: Client;
+
+  type Props = {
+    /**
+     * Test passthrough API clients
+     */
+    api?: Client;
+    /**
+     * Test persistInFlight
+     */
+    persistInFlight?: boolean;
+  };
+
+  const MyComponent: React.FC<Props> = ({api, persistInFlight}) => {
+    apiInstance = useApi({api, persistInFlight});
```

Comment:
imo it's important to test this hook, but it feels like an implementation detail

Reply:
Which part?

Reply:
the whole thing :)

Reply:
I disagree, I think this is a proper test that the api client is provided to the hook caller.

How else would you test a hook like this?

Reply:
I understand and agree with you... I was just thinking that api is an implementation detail that we have to test anyway, but it would be an exception on purpose of the RTL

Reply:
Oh sure, I see what you're saying. Yeah